logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/message/MLSWrongEpochHandler.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.sync.receiver.conversation.message
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logger.KaliumLogger
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.MLSFailure
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.PersistMessageUseCase
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.conversation.JoinExistingMLSConversationUseCase
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.kaliumLogger
+
+interface MLSWrongEpochHandler {
+    suspend fun onMLSWrongEpoch(
+        conversationId: ConversationId,
+        dateIso: String,
+    )
+}
+
+internal class MLSWrongEpochHandlerImpl(
+    private val selfUserId: UserId,
+    private val persistMessage: PersistMessageUseCase,
+    private val conversationRepository: ConversationRepository,
+    private val joinExistingMLSConversation: JoinExistingMLSConversationUseCase
+) : MLSWrongEpochHandler {
+
+    private val logger by lazy { kaliumLogger.withFeatureId(KaliumLogger.Companion.ApplicationFlow.EVENT_RECEIVER) }
+
+    override suspend fun onMLSWrongEpoch(
+        conversationId: ConversationId,
+        dateIso: String,
+    ) {
+        logger.i("Handling MLS WrongEpoch result")
+        conversationRepository.baseInfoById(conversationId).flatMap {
```

Comment:
AFAIK we have a method to get the protocol info only, isn't better to use that since we need the protocol only here?

Reply:
Indeed, it seems we do. I'll change it.